classes/decision/types/BackToInternalReviewFromExternalReview.inc.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+/**
+ * @file classes/decision/types/BackToInternalReviewFromExternalReview.inc.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class decision
+ *
+ * @brief A decision to back out to internal review stage from external review stage
+ */
+
+namespace PKP\decision\types;
+
+use APP\core\Application;
+use APP\decision\Decision;
+use APP\submission\Submission;
+use Illuminate\Validation\Validator;
+use PKP\context\Context;
+use PKP\db\DAORegistry;
+use PKP\decision\DecisionType;
+use PKP\decision\Steps;
+use PKP\decision\steps\Email;
+use PKP\decision\types\contracts\DecisionRetractable;
+use PKP\decision\types\traits\InExternalReviewRound;
+use PKP\decision\types\traits\NotifyAuthors;
+use PKP\decision\types\traits\NotifyReviewers;
+use PKP\decision\types\traits\withReviewRound;
+use PKP\mail\mailables\DecisionBackToInternalFromExternalReviewNotifyReviewer;
+use PKP\mail\mailables\DecisionBackToInternalReviewNotifyAuthor;
+use PKP\security\Role;
+use PKP\submission\reviewAssignment\ReviewAssignmentDAO;
+use PKP\submission\reviewRound\ReviewRound;
+use PKP\user\User;
+
+class BackToInternalReviewFromExternalReview extends DecisionType implements DecisionRetractable
+{
+    use NotifyAuthors;
+    use NotifyReviewers;
+    use withReviewRound;
+    use InExternalReviewRound;
+
+    public function getDecision(): int
+    {
+        return Decision::BACK_TO_INTERNAL_REVIEW_FROM_EXTERNAL_REVIEW;
+    }
+
+    public function getNewStageId(): ?int
+    {
+        return WORKFLOW_STAGE_ID_INTERNAL_REVIEW;
+    }
+
+    public function getNewStatus(): ?int
+    {
+        return null;
+    }
+
+    public function getNewReviewRoundStatus(): ?int
+    {
+        return null;
+    }
+
+    public function getLabel(?string $locale = null): string
+    {
+        return __('editor.submission.decision.backToInternalReviewFromExternalReviewRound', [], $locale);
```

Comment:
I agree it's good to avoid too much inheritance. I would recommend using it, though, where a decision really does mimic another one. I think the internal review stage decisions are a good example, since they mostly mimic the external review stage decisions.

Reply:
reused labels, done .